File: slidingWindow/javascript/twoSum.js
const twoSum = (arr, target) => {
  let result = [];
  let tail = 0,
    sum = 0,
    _twoSum = [];
  const k = 2; // 2-sum

  for (let head = 0; head < arr.length; head++) {
    sum += arr[head];
    result.push(head);
    if (head >= k - 1) {
      if (sum === target) {
        _twoSum.push(result);
        result = [];
      } else {
        result = [head];
      }
      sum -= arr[tail];
      tail += 1;
    }
  }
  return _twoSum;
};
console.log(twoSum([11, -2, 15, 2, 2, 5, 8, 2, 7, 6, 3], 9));

File: slidingWindow/python/twoSum.py
def twoSum(arr, target):
    result = [];
    tail = 0;
    _sum = 0;
    two_sum = [];
    k=2;

    for i in range(len(arr)):
        _sum += arr[i];
        result.append(i);
        if (i >= k - 1):
            if (_sum == target):
                two_sum.append(result);
                result=[]
            else:
                result = [i];
            _sum -= arr[tail];
            tail += 1;
    return two_sum

def main():
    print(str(twoSum([11, -2, 15, 5, 8, 2, 7, 6, 3], 9)))
main()